src/main/kotlin/com/chattriggers/ctjs/triggers/ClassTrigger.kt

```diff
@@ -0,0 +1,62 @@
+package com.chattriggers.ctjs.triggers
+
+import com.chattriggers.ctjs.engine.ILoader
+import com.chattriggers.ctjs.minecraft.wrappers.entity.Entity
+import com.chattriggers.ctjs.minecraft.wrappers.entity.TileEntity
+import com.chattriggers.ctjs.utils.kotlin.MCEntity
+import com.chattriggers.ctjs.utils.kotlin.MCTileEntity
+import net.minecraft.network.Packet
+
+class ClassTrigger(method: Any,private val triggerType: TriggerType, loader: ILoader): Trigger(method, triggerType, loader) {
+    private var triggerClasses: List<Class<*>> = emptyList()
+
+    @Deprecated("Use the better named method", ReplaceWith("setClass(MyClass.class)"))
+    fun setPacketClass(clazz: Class<*>?) = setClasses(listOfNotNull(clazz))
+
+    @Deprecated("Use the better named method", ReplaceWith("setClasses([A.class, B.class, C.class])"))
+    fun setPacketClasses(classes: List<Class<*>>) = setClasses(classes)
+
+    /**
+     * Alias for `setClasses([A.class, B.class])`
+     *
+     * @param clazz The class for which this trigger should run for
+     */
+    fun setClass(clazz: Class<*>?) = setClasses(listOfNotNull(clazz))
```

Comment:
Let's rename the class to `ClassFilterTrigger` and these methods to `setFilteredClass` and `setFilteredClasses`